test/watcher_test.js

```diff
@@ -208,6 +209,7 @@ describe('Watcher', function() {
       await first.resolve();
       await second.resolve();
 
+      await firstBuild;
```

Comment:
Adding this, reproduces the regression that motivated this fix

TODO: leave comment/explicit test so future travelers understand the importance of this.